src/base/settingvalue.h

```diff
@@ -40,17 +39,16 @@
 template <typename T>
 class CachedSettingValue
 {
-    using ProxyFunc = std::function<T (const T&)>;
-
 public:
     explicit CachedSettingValue(const char *keyName, const T &defaultValue = T())
         : m_keyName(QLatin1String(keyName))
         , m_value(loadValue(defaultValue))
     {
     }
 
+    template <typename ProxyFunc>
     explicit CachedSettingValue(const char *keyName, const T &defaultValue
-            , const ProxyFunc &proxyFunc)
+            , ProxyFunc proxyFunc)
```

Comment:
You should keep the original definition here.

Reply:
Excuse me, I don't understand you.

Reply:
AFAIK you are passing a copy of `ProxyFunc` instance, don't you want to pass it by reference?

Reply:
Ah...thanks! No, nor me no you pass anything, because the method is inline.

Reply:
yeah, but semantically you're still taking a copy of a non-POD structure.
FYI: https://stackoverflow.com/questions/722257/should-i-take-arguments-to-inline-functions-by-reference-or-value
I'm too tired for this, I'll leave you be.

Reply:
By using `const` we would disallow mutable lambdas here, that's all. Is that needed?

Reply:
`const` disallows `mutable` lambda specifier.

Reply:
>Is that needed?

Looking at its usage, not now and won't in the future IMO.